packages/mobile/src/screens/track-screen/DownloadSection.tsx

```diff
@@ -52,11 +85,61 @@ export const DownloadSection = ({ trackId }: { trackId: ID }) => {
         justifyContent='space-between'
         alignItems='center'
       >
-        <Flex direction='row' alignItems='center' gap='s'>
-          <IconReceive color='default' />
-          <Text variant='label' size='l' strength='strong'>
-            {messages.title}
-          </Text>
+        <Flex gap='m'>
+          <Flex direction='row' alignItems='center' gap='s'>
+            <IconReceive color='default' />
+            <Text variant='label' size='l' strength='strong'>
+              {messages.title}
+            </Text>
+          </Flex>
+          {shouldDisplayPremiumDownloadLocked && price !== undefined ? (
+            <Button
+              variant='primary'
+              size='small'
+              color='lightGreen'
+              onPress={handlePurchasePress}
+            >
+              {messages.unlockAll(
+                USDC(price / 100).toLocaleString('en-us', {
+                  roundingMode: 'floor',
+                  minimumFractionDigits: 2,
+                  maximumFractionDigits: 2
+                })
+              )}
+            </Button>
+          ) : shouldDisplayPremiumDownloadUnlocked ? (
```

Comment:
super nit: Since `shouldDisplayPremiumDownloadUnlocked` and `shouldDisplayPremiumDownloadLocked` are already mutually exclusive, maybe instead of nested ternary use two separate ones (that each return null in the non-true case)? Open to either, this is somewhat hard to see visually (at least in GitHub) though